integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -500,3 +501,32 @@ def test_greatest(data_gen):
             lambda spark : gen_df(spark, gen).select(
                 f.greatest(*command_args)), conf=allow_negative_scale_of_decimal_conf)
 
+def _assert_div_by_zero(session_func, test_div_func, conf):
+    with pytest.raises(Py4JJavaError) as py4jError:
+        session_func(test_div_func, conf)
+    assert 'java.lang.ArithmeticException: divide by zero' in str(py4jError.value.java_exception)
+
+@pytest.mark.parametrize('ansi_mode', ['nonAnsi', 'ansi'])
+@pytest.mark.parametrize('exp_type', ['bothLiterals', 'justZeroLiteral', 'noLiterals'])
+def test_div_by_zero(ansi_mode, exp_type):
+    if ansi_mode == 'ansi':
+        if is_before_spark_310() or exp_type != 'bothLiterals':
+            pytest.xfail('https://github.com/apache/spark/pull/29882')
```

Comment:
The `exp_type != 'bothLiterals'` logic doesn't make sense to me given the xfail is pointing at a Spark Github issue that is marked as fixed in 3.1.0.  If the latter check doesn't work because of a plugin issue instead of a Spark issue, we should have a different xfail reason pointing at an unresolved plugin issue.  As it is, it's pointing to an issue that is fixed in Spark 3.1.0, so it's not clear why we expect this to fail if we're using Spark 3.1.0.

Reply:
`exp_type != bothLiterals` is documenting that we are inheriting the fix from Spark during constant folding

Reply:
I still don't get why this isn't just `is_before_spark_310`.  IIUC we can't get valid CPU results before Spark 3.1.0, and the plugin should pass this test for Spark >= 3.1.0, so why do we want or need the other check?  Apologies if I'm missing the basics here.